File: TOMIMSV4/Entity/Interface/Pedido_Traslado/Pedido_Traslado_Enc/clsBeI_nav_ped_traslado_enc.vb
Public Class clsBeI_nav_ped_traslado_enc
    Implements ICloneable
    Public Property No() As String = ""
    Public Property Posting_Date() As Date? = Date.Now
    Public Property Receipt_Date() As Date? = Date.Now
    Public Property Shipment_Date() As Date? = Date.Now
    Public Property Status() As Integer
    Public Property Transfer_from_Code() As String = ""
    Public Property Transfer_from_Contact() As String = ""
    Public Property Transfer_from_Name() As String = ""
    Public Property Transfer_to_Code() As String = ""
    Public Property Transfer_to_Contact() As String = ""
    Public Property Transfer_to_Name() As String = ""
    Public Property Transfer_to_CodeField() As String = ""
    Public Property Product_Owner_Code As String = ""
    Public Property Is_Internal_Transfer As Boolean = False
    Public Property Receipt_Document_Reference As String = ""
    Public Property Document_Type As clsDataContractDI.tTipoDocumentoSalida = 0
    Public Property External_Document_No As String = ""
    Public Property RoadCodigoRuta As String = ""
    Public Property RoadCodigoVendedor As String = ""
    Public Property Manufacturing_Process As clsDataContractDI.Manufacturing_Process = 0
    Public Property Address As String = ""
    Public Property Comments As String = ""
    Public Property Company_Code As String = ""
    Public Property IsExport As Boolean = False
    Sub New()
    End Sub
    Sub New(ByRef No As String, ByVal Posting_Date As Date, ByVal Receipt_Date As Date, ByVal Shipment_Date As Date, ByVal Status As Boolean, ByVal Transfer_from_Code As String, ByVal Transfer_from_Contact As String, ByVal Transfer_from_Name As String, ByVal Transfer_to_Code As String, ByVal Transfer_to_Contact As String, ByVal Transfer_to_Name As String, ByVal transfer_to_CodeField As String)
        Me.No = No
        Me.Posting_Date = Posting_Date
        Me.Receipt_Date = Receipt_Date
        Me.Shipment_Date = Shipment_Date
        Me.Status = Status
        Me.Transfer_from_Code = Transfer_from_Code
        Me.Transfer_from_Contact = Transfer_from_Contact
        Me.Transfer_from_Name = Transfer_from_Name
        Me.Transfer_to_Code = Transfer_to_Code
        Me.Transfer_to_Contact = Transfer_to_Contact
        Me.Transfer_to_Name = Transfer_to_Name
        Me.Transfer_to_CodeField = transfer_to_CodeField
    End Sub
    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Pedido/clsBeI_nav_ped_traslado_enc.cs
using static WMS.EntityCore.clsDataContractDI;

namespace WMS.EntityCore.Pedido
{    
    public class clsBeI_nav_ped_traslado_enc : ICloneable
    {
        public string No { get; set; } = string.Empty;
        public DateTime? Posting_Date { get; set; } = DateTime.Now;
        public DateTime? Receipt_Date { get; set; } = DateTime.Now;
        public DateTime? Shipment_Date { get; set; } = DateTime.Now;
        public int Status { get; set; }
        public string Transfer_from_Code { get; set; } = string.Empty;
        public string Transfer_from_Contact { get; set; } = string.Empty;
        public string Transfer_from_Name { get; set; } = string.Empty;
        public string Transfer_to_Code { get; set; } = string.Empty;
        public string Transfer_to_Contact { get; set; } = string.Empty;
        public string Transfer_to_Name { get; set; } = string.Empty;
        public string Transfer_to_CodeField { get; set; } = string.Empty;
        public string Product_Owner_Code { get; set; } = string.Empty;
        public bool Is_Internal_Transfer { get; set; } = false;
        public string Receipt_Document_Reference { get; set; } = string.Empty;
        public tTipoDocumentoSalida Document_Type { get; set; } = 0;
        public string External_Document_No { get; set; } = string.Empty;
        public string RoadCodigoRuta { get; set; } = string.Empty;
        public string RoadCodigoVendedor { get; set; } = string.Empty;
        public clsDataContractDI.Manufacturing_Process Manufacturing_Process { get; set; } = 0;
        public string Address { get; set; } = string.Empty;
        public string Comments { get; set; } = string.Empty;
        public string Company_Code { get; set; } = string.Empty;
        public bool IsExport { get; set; } = false;

        // Nueva propiedad agregada
        public List<clsBeI_nav_ped_traslado_det> Lineas_Detalle { get; set; } = new List<clsBeI_nav_ped_traslado_det>();

        public clsBeI_nav_ped_traslado_enc() { }

        public clsBeI_nav_ped_traslado_enc(
            ref string No,
            DateTime Posting_Date,
            DateTime Receipt_Date,
            DateTime Shipment_Date,
            bool Status,
            string Transfer_from_Code,
            string Transfer_from_Contact,
            string Transfer_from_Name,
            string Transfer_to_Code,
            string Transfer_to_Contact,
            string Transfer_to_Name,
            string transfer_to_CodeField)
        {
            this.No = No;
            this.Posting_Date = Posting_Date;
            this.Receipt_Date = Receipt_Date;
            this.Shipment_Date = Shipment_Date;
            this.Status = Status ? 1 : 0;
            this.Transfer_from_Code = Transfer_from_Code;
            this.Transfer_from_Contact = Transfer_from_Contact;
            this.Transfer_from_Name = Transfer_from_Name;
            this.Transfer_to_Code = Transfer_to_Code;
            this.Transfer_to_Contact = Transfer_to_Contact;
            this.Transfer_to_Name = Transfer_to_Name;
            this.Transfer_to_CodeField = transfer_to_CodeField;
        }

        public object Clone()
        {
            return MemberwiseClone();
        }
    }

}
